### ISSUE TITLE
Unexpected behavior: Animation can be resumed after kill() when pause() is called in between

### ISSUE THREAD

**Issue by CHENJIAMIAN** [greensock/GSAP#594]
I've noticed an unexpected behavior in GSAP animations that might be a bug. Here's the scenario:

1. Create a GSAP animation using fromTo()
2. Call kill() on the animation
3. Call pause() on the animation
4. Call resume() on the animation

Expected behavior:
The animation should not resume, as it has been killed and should no longer exist in the GSAP system.

Actual behavior:
The animation resumes and plays as if it was never killed.

This behavior is inconsistent because:
- If you call resume() immediately after kill(), the animation does not resume (which is expected).
- However, if you insert a pause() call between kill() and resume(), the animation mysteriously comes back to life.

Steps to reproduce:
1. Create a simple GSAP animation:
   ```javascript
   let tween = gsap.fromTo(".myElement", {x: 0}, {x: 100, duration: 2});
   ```
2. Immediately kill the animation:
   ```javascript
   tween.kill();
   ```
3. Then pause it:
   ```javascript
   tween.pause();
   ```
4. Finally, try to resume it:
   ```javascript
   tween.resume();
   ```

The animation will unexpectedly play.

Questions:
1. Is this intended behavior?
2. If not, is this a known issue?
3. Are there any workarounds or best practices to avoid this unexpected behavior?

Environment:
- GSAP version: https://cdnjs.cloudflare.com/ajax/libs/gsap/3.12.4/gsap.min.js

Thank you for your time and consideration!

**Comment by jackdoyle** [comment]
Yes, that's expected behavior although I can see why it could be surprising for some. I'm reluctant to change the behavior at this point because it might break various codebases that rely on the current behavior. It's odd to me that someone would kill() an animation and then later resume() it. Perhaps in version 4.0 we'll alter the behavior because that's a time when people would expect breaking changes. Sorry if this caused any confusion for you. 

As for workarounds, I suppose you could invalidate() and set the vars to an empty object: 
```javascript
tween.kill();
tween.vars = {};
tween.invalidate();
```
Or just add conditional logic to your own codebase, flagging a tween object as "killed" like tween.killed = true and then don't resume() it if that condition is met. There are a lot of options actually for working around this.
